src/fu-main.c

```diff
@@ -563,6 +567,33 @@ fu_main_device_id_valid (const gchar *device_id, GError **error)
 	return FALSE;
 }
 
+/* stolen from ConsoleKit2 */
+static gboolean
+get_trusted (FuMainPrivate *priv, const char *sender)
+{
+	GVariant *value = NULL;
+	uid_t calling_uid;
+	g_autoptr(GError) error_local = NULL;
```

Comment:
I'd be much happier handling the error in the caller, even if that's just a g_warning there...